### ISSUE TITLE
Want a better solution than $y= a_0+a_1x+a_2sin(x)$

### ISSUE THREAD

**Issue by soumitrapy** [sudarsun/DA5401-2024#1]
Hi Sir,
As you were saying in the class, if we use the model $y= a_0+a_1x+a_2sin(x)$ then the model is not perfect. like in the extrapolation case I got something like this:
![Screenshot from 2024-08-18 15-21-42](https://github.com/user-attachments/assets/38532ba0-ea75-4e13-a13d-de9de51a1d95)
Though it was perfect in the initial values, but when x increases, predicted graph is fluctuating from the true graph. Can we fix this only using Linear Regression? by the way, what is the true y function you take?

Similar thing happening with the 2nd curve too [more clearly].
![Screenshot from 2024-08-18 15-38-36](https://github.com/user-attachments/assets/8c799bea-2b71-4c60-a0d4-40bbf399079a)
Here I used $y= a_0+a_1e^{-\lambda x}\sin(3\pi x/94)$. What is the true y function?

**Comment by sudarsun** [comment]
The correct model should be $a_0 + a_1 x + a_2 \sin(a_3 x)$.  Note $a_3$ controlling the periodicity.  Given the error-surface is not convex (refer to https://github.com/sudarsun/DA5401-2024/blob/main/NonConvexLossOfSinusoidModels.ipynb), OLS will not work as you would expect.  You must try an iterative solution such as Gradient descent, to find the best parameters.

**Comment by soumitrapy** [comment]
Suppose I want to use linear regression without without gradient descent. Then I have to choose $a_3$ as a hyper-parameter right? ( is there any way without this?)

**Comment by sudarsun** [comment]
OLS and GD can learn $a_3$, but if you want to keep it simple, you may consider $a_3$ as a hyperparameter.

**Comment by soumitrapy** [comment]
Ok, What about the spring's case? What's the original function?
